### PR TITLE
feat: New tyepscript sandboxes

### LINKED ISSUE #27: Create a TypeScript `@carbon/react` stackblitz sandbox
All the existing sandboxes for `@carbon/react` don't use TypeScript. This repo needs updated with one new sandbox that supports TypeScript for bug reproduction purposes and testing the work happening in  https://github.com/carbon-design-system/carbon/issues/12513 to add TypeScript types to `@carbon/react` exports.

This was asked for in https://github.com/carbon-design-system/carbon/issues/16181

Requirements:
- [ ] Sandbox uses React 18
- [ ] Sandbox uses the same version of typescript that the Carbon monorepo uses (4)
- [ ] Sandbox uses `typescript-config-carbon`
- [ ] Potentially use `eslint-config-carbon`
  - An eslint config is not necessary for the sandbox, but could be used if it makes the authoring experience easier or bugs more readily visible when opening sandboxes.

This work could begin with the Stackblitz [React + Typescript + Vite starter](https://stackblitz.com/edit/vitejs-vite-ntd2xb?file=package.json&terminal=dev).

### PR DESCRIPTION
**Pull request by Gururajj77**
closes #27 

### Added: 
- New folder `react-18-vite-ts` which contains `React 18` with vite buildpack configured with `typescript` version 4 (same as carbon monorepo)
- Added `typescript-config-carbon` to package.json file
- Modified the Readme.md to accommodate the new stackblitz url, please let me know if any modifications required for the link or the placement of the link.

### CODE REVIEW

File: react/latest/react-18-vite-ts/package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  "scripts": {
+    "dev": "vite",
+    "build": "tsc && vite build",
+    "lint": "eslint . --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
+    "preview": "vite preview"
+  },
+  "dependencies": {
+    "@carbon/react": "^1.56.0",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
```
Comment (tay1orjones) [inline react/latest/react-18-vite-ts/package.json:15]:
```suggestion
    "@carbon/react": "^1.x",
    "react": "^18.x",
    "react-dom": "^18.x"
```

File: react/latest/react-18-vite-ts/package.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    "sass": "^1.77.0",
+    "typescript-config-carbon": "^0.2.0"
```
Comment (tay1orjones) [inline react/latest/react-18-vite-ts/package.json:17]: I think these can be moved to devDependencies
```suggestion
```

File: react/latest/react-18-vite-ts/package.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  "devDependencies": {
+    "@types/react": "^18.2.74",
+    "@types/react-dom": "^18.2.24",
+    "@typescript-eslint/eslint-plugin": "^7.5.0",
+    "@typescript-eslint/parser": "^7.5.0",
+    "@vitejs/plugin-react": "^4.2.1",
+    "eslint": "^8.57.0",
+    "eslint-plugin-react-hooks": "^4.6.0",
+    "eslint-plugin-react-refresh": "^0.4.6",
+    "typescript": ">=4.0.0",
```
Comment (tay1orjones) [inline react/latest/react-18-vite-ts/package.json:28]:
```suggestion
    "typescript": ">=4.0.0",
    "sass": "^1.77.0",
    "typescript-config-carbon": "^0.2.0",
```

**Review by tay1orjones — APPROVED** [review]
Just a couple small things, I'll fix em and we can merge this

File: .gitignore [hunk 4]
```diff
@@ -76,6 +76,9 @@ web_modules/
 yarn.lock
 .yarn
 
+# npm lockfile
+package-lock.json
+
```
Comment (tay1orjones) [inline .gitignore:81]: This was the only other thing - the sandboxes don't need lockfiles. They should always use the most recent version of the packages specified in the package.json

**Review by tay1orjones — APPROVED** [review]
I've done some general cleanup, I think this is good to go

**Comment by Gururajj77** [comment]
Thanks for the cleanup! will learn a lot from this!!

**Pull request merged**